bolt-modules/boltlib/lib/puppet/functions/get_resources.rb

```diff
@@ -23,11 +23,15 @@
   end
 
   def script_compiler
-    @script_compiler ||= Puppet::Pal::ScriptCompiler.new(closure_scope.compiler)
+    @script_compiler ||= Puppet.lookup(:pal_script_compiler)
   end
 
   def run_task(executor, targets, name, args = {})
-    tasksig = script_compiler.task_signature(name)
+    begin
+      tasksig = script_compiler.task_signature(name)
```

Comment:
Should we put a method somewhere that accepts a compiler and a task name and then does this? It feels weird doing this same begin/rescue everywhere, but on the other hand where would the method live that makes sense all the places this is used?

Reply:
I think having a wrapper in the Bolt::PAL class to do this makes sense. Something like:
```
def self.task_signature(name)
  script_compiler = Puppet.lookup(:pal_script_compiler)
  begin
    tasksig = script_compiler.task_signature(name)
  rescue
    ...
  end
  ...
```

I haven't looked at the other functions but it seems like this is the only place we use the script_compiler, so we could remove that entirely from at least this function.

Reply:
I'm also not sure how looking up the script compiler is different from using the compiler configured for Bolt?
https://github.com/puppetlabs/bolt/blob/master/lib/bolt/pal.rb#L136-L137

Reply:
So if task_signature (with is already an instance method in pal) were to be a class method, boltlib functions would require bolt/pal? Is that right?

Reply:
Should be unnecessary given upstream validation.